_posts/2017-02-02-overlay-network-performance-testing.md

```diff
@@ -0,0 +1,75 @@
+
+# What are the options?
+It turns out there are a wide variety of overlay networks to choose from and they all have different characteristics.  The current networking model for containers (e.g. Docker and Rkt - CoreOS' similar product) is called CNI and theoretically any overlay network that is CNI compliant can be used in Kubernetes.  The first wave of cutting down these networks is by ease of install - If I'm going to be testing networks en masse, I don't want to be spending days installing to find they're not suitable.
+
+This meant a much smaller list to benchmark - the ones that had kubeadm addons that can be installed quickly on a disposable cluster.  This list is :
+
+* Flannel - the default CoreOS (what we're probably going to use as our base OS) overlay network
+* Calico - the first major overlay network to have network policy support (ACLs for network, similar to firewalls)
+* Canal - This is effectively Flannel's overlay network with Calico's policy model, and is so similar to Flannel it's probably not worth testing
+* Weave - Another good alternative that has network policy support built in
+* Romana - This turned out to be not an overlay network per se, although it is a networking layer, but was later discarded because we want an overlay network
+
+# Requirements other than performance
+
+* **Sky Betting and Gaming** exists in a regulated business so we do require our new platform to have network policy support from the get go
+* We also have to be able to explain the stack to all our on-call engineers, so the simpler the better
+* Our newest datacentre is being set up with Cisco's ACI kit so support for that would be nice to have, but not in any way a necessity
+* The network must work in AWS as that is where our initial deployment will be (it can't interfere with AWS' networking)
+
+It turns out, other than the ACI nice to have, the three we whittled it down to all cover those if you add the Calico policy model to Flannel.
+
+# How do you test network performance
+I have learned everything I know about networking by osmosis and doing.  This puts me at a disadvantage when it comes to network performance because that's usually "Somebody else's problem".  Not this time, this time it's mine, oh well, a learning opportunity.
+
+Having done a fair bit of reading and looking at the tools available to test the various networks, I settled on the fine tool [iperf3](https://iperf.fr/) which specifically allows us to measure using UDP and measure [jitter](https://en.wikipedia.org/wiki/Packet_delay_variation) (which I knew to be a bad thing, but didn't have a good grip on what it was before reading about it).
+
+To ensure that the test was cross-AZ in amazon (profile for worst-case, it can only get better!), I used a container replica set of 3 on a kubernetes 3-worker cluster.
```

Comment:
It's not clear to me what "cross-AZ" means in this sentence.  Consider rephrasing.